junit-jupiter-engine/src/test/java/org/junit/jupiter/api/AssertTimeoutAssertionsTests.java

```diff
@@ -158,11 +153,11 @@ void assertTimeoutWithMessageSupplierForSupplierThatCompletesAfterTheTimeout() {
 	 * Take a nap for 100 milliseconds.
 	 */
 	private void nap() throws InterruptedException {
-		long start = System.currentTimeMillis();
+		long start = System.nanoTime();
 		// workaround for imprecise clocks (yes, Windows, I'm talking about you)
 		do {
 			Thread.sleep(100);
-		} while (System.currentTimeMillis() - start < 100);
+		} while (System.nanoTime() - start < 100_000_000L);
```

Comment:
Since this is not related to `throwAsUncheckedException()`, please move these changes to a separate PR with the rationale for the change.

Reply:
Reverted.